engine/render/renderer/platform/vulkan/Fence.h

```diff
@@ -0,0 +1,128 @@
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_FENCE_H
+#define SIEGE_ENGINE_FENCE_H
+
+#include "utils/Types.h"
+
+namespace Siege::Vulkan
+{
+/**
+ * The Fence class is a thin wrapper around the Vulkan VkFence type. This type supports both single
+ * and multiple Fence instances (such as frame synchronisation). By default, the fences are
+ * initialised as null values
+ */
+class Fence
+{
+public:
+
+    /**
+     * The initial state of the Fence
+     */
+    enum FenceInitState
+    {
+        FENCE_EMPTY = 0,
+        FENCE_NULL = 1
+    };
+
+    /**
+     * Zero-value constructor. Initialises the Fence object with null values to begin with
+     */
+    Fence() : size {0}, initState {FENCE_NULL} {}
+
+    /**
+     * Initialises the Fence class. Can be initiated to store a contiguous block of fences, or can
+     * be used to create only a single fence
+     * @param fenceCount the number of fences to be initialised
+     * @param initState the initialisation state - can either be NULL or EMPTY
+     */
+    Fence(size_t fenceCount, FenceInitState = FENCE_EMPTY);
+
+    /**
+     * Move constructor for the Fence class
+     * @param other the Fence class to be moved
+     */
+    inline Fence(Fence&& other)
+    {
+        Swap(other);
+    };
+
+    /**
+     * Destructor for the Fence class
+     */
+    ~Fence();
+
+    /**
+     * Move assignment operator
+     * @param other the Fence class to be moved
+     * @return a reference to the created Fence
+     */
+    inline Fence& operator=(Fence&& other)
+    {
+        Swap(other);
+        return *this;
+    }
+
+    /**
+     * Subscript operator to index into the stored fences
+     * @param index the index to access
+     * @return the Fence that's stored in the index location
+     */
+    inline VkFence& operator[](size_t index)
+    {
+        return Get(index);
+    }
+
+    /**
+     * Gets the raw vulkan Fence pointer stored by the wrapper
+     * @param index the index to access, defaults to 0
+     * @return the pointer to the raw vulkan pointer
+     */
```

Comment:
Methods like this should be marked up with an `@warn` to let you know it doesn't bounds check

Reply:
Done.